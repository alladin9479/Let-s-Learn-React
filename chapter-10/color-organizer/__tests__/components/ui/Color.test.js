import { shallow, mount } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { compose } from 'redux'
import Color from '../../../src/components/ui/Color'

describe("<Color /> UI Component", () => {

  it("Renders correct properties", () =>
    compose(expect,toJSON,shallow)(
      <Color title="Test Color" color="#F0F0F0" rating={3} />
    ).toMatchSnapshot()
  )

  it("Invokes onRate property", () => {
    mount(<Color title="Test Color" color="#F0F0F0" />)
      .find("div.star-rating")
      .childAt(2)
      .simulate('click')
  })

  it("Invokes onRate property", () => {
    mount(<Color title="Test Color" color="#F0F0F0" />)
      .find("button")
      .simulate('click')
  })

})
