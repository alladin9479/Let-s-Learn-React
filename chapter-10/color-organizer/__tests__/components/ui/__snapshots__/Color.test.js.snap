exports[`<Color /> UI Component Renders correct properties 1`] = `
<section
  className="color">
  <h1>
    Test Color
  </h1>
  <button
    onClick={[Function]}>
    <Component />
  </button>
  <div
    className="color"
    style={
      Object {
        "backgroundColor": "#F0F0F0",
      }
    } />
  <TimeAgo />
  <div>
    <StarRating
      onRate={[Function]}
      starsSelected={3} />
  </div>
</section>
`;
